### ISSUE TITLE
Implementation detail on UCF101 dataset

### ISSUE THREAD

**Issue by fanser** [rohitgirdhar/ActionVLAD#2]
Hi, I want to do some experiments on UCF101 dataset. and I have finished training the stage1  of rgb stream, and I got about 0.78 mean accuracy. so 1. is this a reasonable result?
And when trained the stage1 of flow stream,  the model didn't convergence after 2000 itr  with base_lr=0.01 and other parameters follow your paper. so 2. what's wrong about it?

**Comment by rohitgirdhar** [comment]
1. That sounds about right. I remember getting ~79 for stage 1 and ~83 after stage 2.
2. So flow typically takes longer and doesn't overfit to training as much. Maybe try testing the final model to see if it gives reasonable performance

**Comment by fanser** [comment]
Thanks for your analysis, but i have another questions.
1.How to sort the input flow images?
I use this format in channel map: [flow_x_1, flow_y_1,flow_x_2, flow_y_2,...,flow_x_10, flow_y_10]
or [flow_x_1,flow_x_2,...,flow_x_10, flow_y_1,...,flow_y_2]
2. I  rescale the flow_imgs to 0-1 and then minus mean=0.5 as the input. so Is it the right preprocessing?
3.I found the rgb_stage1 model is over-feating, because the train accuracy is ~0.96 and the test accuracy is ~0.73(sorry i told you ~0.78 before for my bad memory)

**Comment by fanser** [comment]
I figure out the first and second question.
**2. I rescale the flow_imgs to 0-1 and then minus mean=0.5 as the input. so Is it the right preprocessing?**
The answer is wrong. The right preprocessing is input = flow_img - 128 with the flow_img pixel value from 1 to 256( not 0 to1). Because I use the pretrained model from [Temporal UCF101 Split1 | cuhk_action_temporal_vgg_16_split1.caffemodel](https://github.com/yjxiong/caffe/tree/action_recog/models/action_recognition)
And the channel map should be [flow_x_1, flow_y_1,flow_x_2, flow_y_2,...,flow_x_10, flow_y_10]
Finally I believe the flow_stage1 model should be convergence

**Comment by rohitgirdhar** [comment]
1. The first ordering is correct. 
2. You figured it out correctly
3. For UCF101, I trained on top of the UCF 2-stream [pretrained models](https://github.com/yjxiong/caffe/tree/action_recog/models/action_recognition), and not just the imagenet pretrained model as I use for HMDB. That might explain the difference.

**Comment by fanser** [comment]
So the vgg16 pretrained on imagenet is for HMDB, and the UCF 2-stream pretrained models is for UCF101.

**Comment by rohitgirdhar** [comment]
That's correct. Though I always start with the 2-stream initialization, what happened here is that for HMDB I only train the last 2 FC layers in 2-stream, so when I clip that off for the ActionVLAD layer, what's left is the ImageNet initialization.

**Comment by fanser** [comment]
thanks. I have another question about the pretrained model of the flow_stream on HMDB51.
whether fine-tune on the 2-stream or retrain a new flow_stream model on HMDB51?

**Comment by rohitgirdhar** [comment]
Flow stream ActionVLAD is always trained from the 2-stream trained model, for all datasets. That's because flow stream in a 2-stream architecture is trained until conv1.

**Comment by fanser** [comment]
@rohitgirdhar how many fc layers after vladlayer? 
(64x512)->101 or (64x512)->4096->101

**Comment by rohitgirdhar** [comment]
Single layer ((64x512)->101)
